packages/poller/src/helpers/intent.ts

```diff
@@ -581,6 +590,247 @@ export const sendSvmIntents = async (
+
+    try {
+      const approvalResult = await submitTransactionWithLogging({
+        chainService,
+        logger,
+        chainId: originChainId,
+        txRequest: {
+          chainId: +originChainId,
+          to: firstIntent.inputAsset,
+          data: approveData,
+          value: '0',
+          from: tronAddress,
+          funcSig: approveFunctionSig,
+        },
+        zodiacConfig: originWalletConfig,
+        context: { requestId, invoiceId, transactionType: 'trc20-approval' },
+      });
+
+      logger.info('Approval completed for intent batch', {
+        requestId,
+        invoiceId,
+        chainId: originChainId,
+        approvalTxHash: approvalResult.hash,
+        hadZeroApproval: isUSDT,
+        asset: firstIntent.inputAsset,
+        amount: totalAmount.toString(),
+      });
+
+      if (prometheus) {
+        prometheus.updateGasSpent(originChainId, TransactionReason.Approval, BigInt(0));
+      }
+    } catch (error) {
+      logger.error('Failed to approve TRC20 on Tron', {
+        requestId,
+        invoiceId,
+        error,
+        tokenAddress: firstIntent.inputAsset,
+        amount: totalAmount.toString(),
+      });
+      throw error;
+    }
+
+    logger.info(`Processing ${intents.length} total intent(s)`, {
+      requestId,
+      invoiceId,
+      count: intents.length,
+      origin: intents[0].origin,
+      token: intents[0].inputAsset,
+    });
+
+    // Verify min amounts for all intents before sending the batch
+    for (const intent of intents) {
+      // Sanity check -- minAmounts < intent.amount
+      const { minAmounts } = await everclear.getMinAmounts(invoiceId);
+      if (BigInt(minAmounts[intent.origin] ?? '0') < BigInt(intent.amount)) {
+        logger.warn('Latest min amount for origin is smaller than intent size', {
+          minAmount: minAmounts[intent.origin] ?? '0',
+          intent,
+          invoiceId,
+          requestId,
+        });
+        continue;
+        // NOTE: continue instead of exit in case other intents are still below the min amount,
+        // then you would still be contributing to invoice to settlement. The invoice will be handled
+        // again on the next polling cycle.
+      }
+    }
+
+    const purchaseResult = await submitTransactionWithLogging({
+      chainService,
+      logger,
+      chainId: originChainId,
+      txRequest: {
+        chainId: +originChainId,
+        to: feeAdapterTxData.to as string,
+        data: feeAdapterTxData.data,
+        value: feeAdapterTxData.value ?? '0',
+        from: tronAddress,
+        funcSig: 'newIntent(uint32[],address,address,address,uint256,uint24,uint48,bytes,(uint256,uint256,bytes))',
+      },
+      zodiacConfig: originWalletConfig,
+      context: { requestId, invoiceId, transactionType: 'batch-create-intent' },
+    });
+
+    const purchaseTx = purchaseResult.receipt!;
+    const purchaseIntentIds = await getAddedIntentIdsFromReceipt(purchaseTx, intents[0].origin, logger, {
```

Comment:
Tx logs may be empty in the receipt, I'd suggest to fill `purchaseIntentIds` from `intents` argument as for Solana.

Reply:
Though tx logs are not returned immediately, the confirmation loop should fetch the logs. So really not an issue.